pkg/controller/util/endpoint/controller_utils_test.go

```diff
@@ -499,3 +499,165 @@ func BenchmarkGetPodServiceMemberships(b *testing.B) {
 		}
 	}
 }
+
+func Test_podChanged(t *testing.T) {
```

Comment:
I think this function might be easier to understand and less error prone if it used test cases and a modifier function like https://github.com/kubernetes/kubernetes/pull/93089. I know I ran into some significant bugs in the previous test implementation like this where things were passing for reasons that were not intended.